src/torchmetrics/functional/regression/weighted_pearson.py

```diff
@@ -0,0 +1,176 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+
+import torch
+from torch import Tensor
+
+from torchmetrics.functional.regression.utils import _check_data_shape_to_num_outputs, _check_data_shape_to_weights
+from torchmetrics.utilities import rank_zero_warn
+from torchmetrics.utilities.checks import _check_same_shape
+
+
+def _weighted_pearson_corrcoef_update(
+    preds: Tensor,
+    target: Tensor,
+    weights: Tensor,
+    mean_x: Tensor,
+    mean_y: Tensor,
+    var_x: Tensor,
+    var_y: Tensor,
+    cov_xy: Tensor,
+    weights_prior: Tensor,
+    num_outputs: int,
+) -> tuple[Tensor, Tensor, Tensor, Tensor, Tensor, Tensor]:
+    """Update and returns variables required to compute weighted Pearson Correlation Coefficient.
+
+    Check for same shape of input tensors.
+
+    Updates are based on `Algorithms for calculating variance`_. Specifically, `online weighted variance`_ and
+    `online weighted covariance`_.
+
+    Variance intentionally not divided by sum of weights in `update` step as it is computed as necessary in
+    the `compute` step.
+
+    Args:
+        preds: estimated scores
+        target: ground truth scores
+        weights: weight associated with scores
+        mean_x: current mean estimate of x tensor
+        mean_y: current mean estimate of y tensor
+        var_x: current variance estimate of x tensor
+        var_y: current variance estimate of y tensor
+        cov_xy: current covariance estimate between x and y tensor
+        weights_prior: current sum of weights
+        num_outputs: number of outputs in multioutput setting
+
+    """
+    # Data checking
+    _check_same_shape(preds, target)
+    _check_data_shape_to_num_outputs(preds, target, num_outputs)
+    _check_data_shape_to_weights(preds, weights)
+
+    weights_sum = weights.sum()
+
+    if weights_prior > 0:  # True if prior observations exist
+        mx_new = (weights_prior * mean_x + (weights * preds).sum(0)) / (weights_prior + weights_sum)
+        my_new = (weights_prior * mean_y + (weights * target).sum(0)) / (weights_prior + weights_sum)
+
+        var_x += (weights * (preds - mx_new) * (preds - mean_x)).sum(0)
+        var_y += (weights * (target - my_new) * (target - mean_y)).sum(0)
+    else:
+        mx_new = ((weights * preds).sum(0) / weights_sum).to(mean_x.dtype)
+        my_new = ((weights * target).sum(0) / weights_sum).to(mean_y.dtype)
+
+        var_x = (weights * (preds - mx_new) ** 2).sum(0)
+        var_y = (weights * (target - my_new) ** 2).sum(0)
+
+    cov_xy += (weights * (preds - mx_new) * (target - mean_y)).sum(0)
```

Comment:
its a long time since i touched the pearson metric, so just wondering:
is there a reason for the asymmetric update here using `mean_y` instead of `my_new` when the `mx_new`  is used for the x variance

Reply:
This is an error on my part. Updated to use `my_new`.